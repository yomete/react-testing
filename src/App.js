import React, { Component } from 'react';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      status: 0,
    }
  }

  setOn = event => {
    this.setState({status: 1})
  }; 

  setOff = event => {
    this.setState({status: 0})
  }; 

  render() {
    return (
      <div className="App">
        <p className="value">
          {this.state.status}
        </p>

        <button id="offButton" onClick={this.setOff}>Off</button>
        <button id="onButton" onClick={this.setOn}>On</button>
      </div>
    );
  }
}

export default App;
