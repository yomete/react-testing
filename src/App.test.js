import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
var assert = require('assert');

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should return -1 when the value is not present', function() {
  assert.equal([1,2,3].indexOf(4), -1);
});

describe('Enzyme Tests', () => {
  it('App loads with initial state of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toEqual('0');
  });

  it('on button works as expected', () => {
    const wrapper = shallow(<App />);
    const onButton = wrapper.find('button#onButton');
    onButton.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('1');
  });

  it('off button works as expected', () => {
    const wrapper = shallow(<App />);
    const offButton = wrapper.find('button#offButton');
    offButton.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('0');
  });
});

describe('Jest Tests', () => {
  it('snapshot matches', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
