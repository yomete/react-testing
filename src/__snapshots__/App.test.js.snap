// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Tests snapshot matches 1`] = `
<div
  className="App"
>
  <p
    className="value"
  >
    0
  </p>
  <button
    id="offButton"
    onClick={[Function]}
  >
    Off
  </button>
  <button
    id="onButton"
    onClick={[Function]}
  >
    On
  </button>
</div>
`;
